envi/archs/arm/disasm.py

```diff
@@ -1350,25 +1350,7 @@ def getBranches(self, emu=None):
         if self.prefixes != COND_AL:
             flags |= envi.BR_COND
 
-        if self.opcode in ( INS_B, INS_BX ):
-            oper = self.opers[0]
-
-            # check for location being ODD
-            operval = oper.getOperValue(self)
-            if operval == None:
-                return ret
-
-            if self.opcode == INS_BX and operval & 3:
-                flags |= envi.ARCH_THUMB16
-            # if we don't know that it's thumb, default to "ARCH_DEFAULT"
-            #else:
-            #    flags |= envi.ARCH_ARMV7
-
-
-            operval &= 0xfffffffe           # this has to work for both arm and thumb
-            ret.append((operval, flags))
-
-        elif self.opcode in ( INS_BL, INS_BLX ):
+        if self.opcode in ( INS_B, INS_BX, INS_BL, INS_BLX, INS_BCC ):
```

Comment:
merge this together to tighten up the code, and handle the shift for arm/thumb better.